pommerman/network/ion_server/network.py

```diff
@@ -0,0 +1,205 @@
+    """Decode the messages recieved from the clients"""
+    if message["i"] == nic.check.value:
+        await websocket.send(
+            dumps({
+                "i": nic.status_ok.value,
+                "p": len(PLAYER_WS),
+                "m": len(MATCH_PROCESS)
+            }))
+    elif message["i"] == nic.match_act.value:
+        if message["sid"] == MATCH_PROCESS[message["u"]]["id"]:
+            MATCH_PROCESS[message["u"]]["act"][MATCH_PROCESS[message["u"]][
+                "pl"].index(message["id"])] = message["act"]
+            MATCH_PROCESS[message["u"]]["recv"][MATCH_PROCESS[message["u"]][
+                "pl"].index(message["id"])] = True
+    elif message["i"] == nic.replay.value:
+        try:
+            with open(
+                    path.join(
+                        path.join(getcwd(), "matches"),
+                        str(message["p"]) + ".json"), 'r') as f:
+                # Note: Registry expression match comes after as it's an expensive operation as compared to file I/O
+                if fullmatch("^[a-z0-9-]*$", message["p"]) is not None:
+                    f = load(f)
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_ok.value, f]), "utf8")))
+                else:
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+        except:
+            await websocket.send(
+                compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+    elif len(PLAYER_WS) < MAX_PLAYERS and (message["i"] == nic.match.value
+                                           or message["i"] == nic.room.value):
+        uuid = str(uuid4())
+        while uuid in PLAYER_WS:
+            uuid = str(uuid4())
+        PLAYER_WS[uuid] = {"ws": websocket}
+        if message["i"] == nic.match.value:
+            CONCURRENTLY_LOOKING["noroom"].append(uuid)
+            PLAYER_WS[uuid]["l"] = True
+        elif message["i"] == nic.room.value:
+            if message["room"] in CONCURRENTLY_LOOKING["room"]:
+                if len(CONCURRENTLY_LOOKING["room"][message["room"]]) < 4:
+                    CONCURRENTLY_LOOKING["room"][message["room"]].append(uuid)
+            else:
+                CONCURRENTLY_LOOKING["room"][message["room"]] = [uuid]
+            PLAYER_WS[uuid]["l"] = False
+            PLAYER_WS[uuid]["r"] = str(message["room"])
+        await websocket.send(dumps({"i": nic.status_reg.value, "id": uuid, "m": MODE}))
+
+
+async def ws_handler(websocket, pth=None): # pylint: disable=unused-argument
+    """Handle the messages recieved by WebSocket"""
+    global THREAD_LOCK
+    while THREAD_LOCK:
+        pass
+    THREAD_LOCK = True
+    async for message in websocket:
+        try:
+            message = loads(message)
+        except JSONDecodeError:
+            pass
+        try:
+            decode_message(message, websocket)
+        except:
+            pass
```

Comment:
I'm confused by this function. It:
1. tries to load the message.
2. if it fails to do that, it still tries to decode the message. 
3. If it fails to do that, it passes.

A) Why should it expect to be able to decode the message if it didn't load? 
B) Does anyone expect it to say something if it failed to decode?

Reply:
A) That didn't make sense. I've put them under the same try statement.
B) No, it doesn't actually the decode fail will most likely result in the function failing aswell as it's dependant on that.
Done 👍

Reply:
Keep unresolved pls.